extension/toolsnextbot.cpp

```diff
@@ -1,7 +1,198 @@
+	RECONFIGURE("CBaseCombatCharacter::OnNavAreaChanged", ToolsNextBot_OnNavAreaChanged)
+	RECONFIGURE("CBaseEntity::Touch", ToolsNextBot_Touch)
+	RECONFIGURE("CBaseCombatCharacter::Weapon_Equip", ToolsNextBot_Weapon_Equip)
+	RECONFIGURE("CBaseCombatCharacter::Weapon_Drop", ToolsNextBot_Weapon_Drop)
+
+	return (NextBotManager::Init(config, error, maxlength)
+		&& INextBotComponent::Init(config, error, maxlength)
+		&& ILocomotion::Init(config, error, maxlength)
+		&& NextBotCombatCharacter::Init(config, error, maxlength)
+		&& NextBotGroundLocomotion::Init(config, error, maxlength));
+}
+
 ToolsNextBot::ToolsNextBot(CBaseCombatCharacterHack* link) :
 	INextBot(),
 	m_linkedEntity(link)
 {
+}
+
+ToolsNextBotPlayer::ToolsNextBotPlayer(CBaseCombatCharacterHack* link) :
+	ToolsNextBot(link)
+{
+	m_burningTimer.Invalidate();
+
+	m_hooks.push_back(SH_ADD_MANUALHOOK(ToolsNextBot_Spawn, link, SH_MEMBER(this, &ToolsNextBotPlayer::Hook_Spawn), false));
+	m_hooks.push_back(SH_ADD_MANUALHOOK(ToolsNextBot_PhysicsSimulate, link, SH_MEMBER(this, &ToolsNextBotPlayer::Hook_PhysicsSimulate), false));
+	m_hooks.push_back(SH_ADD_MANUALHOOK(ToolsNextBot_OnTakeDamage_Alive, link, SH_MEMBER(this, &ToolsNextBotPlayer::Hook_OnTakeDamage_Alive), false));
+	m_hooks.push_back(SH_ADD_MANUALHOOK(ToolsNextBot_OnTakeDamage_Dying, link, SH_MEMBER(this, &ToolsNextBotPlayer::Hook_OnTakeDamage_Dying), false));
+	m_hooks.push_back(SH_ADD_MANUALHOOK(ToolsNextBot_Event_Killed, link, SH_MEMBER(this, &ToolsNextBotPlayer::Hook_Event_Killed), false));
+	m_hooks.push_back(SH_ADD_MANUALHOOK(ToolsNextBot_HandleAnimEvent, link, SH_MEMBER(this, &ToolsNextBotPlayer::Hook_HandleAnimEvent), false));
+	m_hooks.push_back(SH_ADD_MANUALHOOK(ToolsNextBot_OnNavAreaChanged, link, SH_MEMBER(this, &ToolsNextBotPlayer::Hook_OnNavAreaChanged), false));
+	m_hooks.push_back(SH_ADD_MANUALHOOK(ToolsNextBot_Touch, link, SH_MEMBER(this, &ToolsNextBotPlayer::Hook_Touch), false));
+	m_hooks.push_back(SH_ADD_MANUALHOOK(ToolsNextBot_Weapon_Equip, link, SH_MEMBER(this, &ToolsNextBotPlayer::Hook_Weapon_Equip), false));
+	m_hooks.push_back(SH_ADD_MANUALHOOK(ToolsNextBot_Weapon_Drop, link, SH_MEMBER(this, &ToolsNextBotPlayer::Hook_Weapon_Drop), false));
+}
+
+ToolsNextBotPlayer::~ToolsNextBotPlayer()
+{
+	for (auto it = m_hooks.begin(); it != m_hooks.end(); it++)
+	{
+		SH_REMOVE_HOOK_ID((*it));
+	}
+}
+
+void ToolsNextBotPlayer::Update()
+{
+	if ((GetEntity()->IsAlive() || !IsDormantWhenDead()) && !NextBotPlayerStop->GetBool())
+	{
+		ToolsNextBot::Update();	
+	}
+}
+
+bool ToolsNextBotPlayer::IsRemovedOnReset() const
+{
+	return false;
+}
+
+void ToolsNextBotPlayer::Hook_Spawn(void)
+{
+	m_burningTimer.Invalidate();
+
+	ToolsNextBot::Reset();
+	RETURN_META(MRES_IGNORED);
+}
+
+void ToolsNextBotPlayer::Hook_PhysicsSimulate(void)
+{
+	VPROF("ToolsNextBotPlayer::PhysicsSimulate");
+
+	if ((IsDormantWhenDead() && GetEntity()->GetLifeState() == LIFE_DEAD) || NextBotStop->GetBool())
```

Comment:
This was already discussed before in Discord, but having `IsDormantWhenDead()` return true will cause issues when it comes to invoking `NextBotAction` changes upon an `OnKilled` event. In short, event changes from an `OnKilled` event will not happen because event changes are not applied on a `NextBotAction` until the *next* update is invoked.

https://github.com/TF2-DMB/CBaseNPC/blob/4cb7a1b5873b8b96c7d0f7b033e97b29bf253355/extension/sourcesdk/NextBot/NextBotBehavior.h#L251

Because `IsDormantWhenDead()` returns true, this halts any more updates from happening on the Behavior interface, resulting in no change. Therefore, there should be a method/property that allows for overriding `IsDormantWhenDead()` to address this issue.